lib/std/tar.zig

```diff
@@ -699,6 +723,22 @@ fn createDirAndSymlink(dir: std.fs.Dir, link_name: []const u8, file_name: []cons
     };
 }
 
+// Creates a hard link at path `file_name` which points to `link_name`.
+fn createDirAndHardlink(dir: std.fs.Dir, link_name: []const u8, file_name: []const u8) !void {
+    if (native_os == .windows) {
+        return error.UnableToCreateHardLink;
+    }
```

Comment:
https://learn.microsoft.com/en-us/windows/win32/fileio/hard-links-and-junctions

Not sure how compatible Windows's notion of a hard link is with the POSIX notion, though. cc @squeek502

Reply:
Out of precaution and without a concrete example, I would not assume they work in the same way

Reply:
I'd say open a follow-up issue for hard links on Windows. I'm not familiar enough with hard links to know how Windows/POSIX hard links differ.